httpclient5/src/main/java/org/apache/hc/client5/http/ssl/DefaultHostnameVerifier.java

```diff
@@ -228,8 +229,19 @@ private static boolean matchIdentity(final String host, final String identity,
                                          final PublicSuffixMatcher publicSuffixMatcher,
                                          final DomainType domainType,
                                          final boolean strict) {
-        if (publicSuffixMatcher != null && host.contains(".")) {
-            if (publicSuffixMatcher.getDomainRoot(identity, domainType) == null) {
+
+        final String normalizedHost;
+        final String normalizedIdentity;
+        try {
+            normalizedHost = IDN.toUnicode(host);
```

Comment:
@arturobernalg This is unnecessary. The hostname has already been normalized. Please see `Host` and `HttpHost` classes

Reply:
@ok2c 
you right. changed. Please do another pass